ces_admin/views.py

```diff
@@ -22,6 +33,80 @@ def ces_logout(request):
     logout(request)
     return HttpResponseRedirect('/')
 
-@login_required(login_url='/ces-login/')
-def ces_admin(request):
-    return render(request, 'ces_admin/ces-admin.html')
\ No newline at end of file
+class DashboardView(TemplateView):
+    template_name = 'ces_admin/ces-dashboard.html'
+    
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        
+        base_query = '''
```

Comment:
Any particular reason for not using the ORM here?

Reply:
I don't really have a problem with the sql for complicated join, but the using string formatting to alter the where clause kind of feels weird to me. It seems like you are kind of doing some of what an ORM does already with that.

It would be nice if it was possible filter raw query sets, but that doesn't look possible. https://stackoverflow.com/questions/9135678/django-filter-a-rawqueryset

Given that, this seems like a pretty sane way of doing it.

You probably do want to find a more informative name than "base_query"